lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -227,6 +227,8 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._contentType = privateData.options.contentType;
+    this._encoding = privateData.options.encoding;
```

Comment:
This may require implementing a copy of https://encoding.spec.whatwg.org/#names-and-labels or something similar, that maps all the possible encoding labels (which I think is what you call characterSet) to iconv-lite-compatible strings.